bundles/org.openhab.core.config.discovery.addon/src/main/java/org/openhab/core/config/discovery/addon/AddonSuggestionService.java

```diff
@@ -112,49 +94,56 @@ protected void removeAddonFinderService(AddonFinderService addonFinderService) {
+                    if (currentEnabled != newEnabled) {
+                        String type = SUGGESTION_FINDER_TYPES.get(finder);
+                        if (type != null) {
+                            logger.debug("baseFinderConfig {} {} = {} => updating from {} to {}", finder, cfgParam,
+                                    config.get(cfgParam), currentEnabled, newEnabled);
+                            baseFinderConfig.put(finder, newEnabled);
+                            if (finderService != null) {
+                                if (newEnabled) {
+                                    finderService.install(type);
+                                } else {
+                                    finderService.uninstall(type);
+                                }
+                            }
                         } else {
-                            finderService.uninstall(type);
+                            logger.warn("Failed to resolve addon suggestion finder type for suggestion finder {}",
+                                    finder);
                         }
                     }
                 }
-            }
-        });
-        this.config = config;
+            });
+        }
     }
 
-    private void syncConfiguration() {
-        final Map<String, Object> cfg = getConfiguration();
-        if (cfg != null && !cfg.equals(config)) {
-            modified(cfg);
+    private void initAddonFinderService() {
+        AddonFinderService finderService = this.addonFinderService;
+        if (finderService == null) {
+            return;
         }
-    }
 
-    private @Nullable Map<String, Object> getConfiguration() {
-        try {
-            Dictionary<String, Object> cfg = configurationAdmin.getConfiguration(CONFIG_PID).getProperties();
-            if (cfg != null) {
-                List<String> keys = Collections.list(cfg.keys());
-                return keys.stream().collect(Collectors.toMap(Function.identity(), cfg::get));
+        String type;
+        for (Entry<String, Boolean> entry : baseFinderConfig.entrySet()) {
+            type = SUGGESTION_FINDER_TYPES.get(entry.getKey());
+            if (type != null) {
+                if (entry.getValue() instanceof Boolean enabled) {
+                    if (enabled) {
+                        finderService.install(type);
+                    } else {
+                        finderService.uninstall(type);
+                    }
+                }
+            } else {
+                logger.warn("Failed to resolve addon suggestion finder type for suggestion finder {}", entry.getKey());
             }
-        } catch (IOException | IllegalStateException e) {
-            logger.debug("Exception occurred while trying to get the configuration: {}", e.getMessage());
         }
-        return null;
     }
 
     private boolean isFinderEnabled(AddonFinder finder) {
```

Comment:
```suggestion
    private boolean isFinderEnabled(AddonFinder finder) {
    if (addonFinderService == null) {
        return false;
    }
```

Reply:
@lolodomo I mean something like this.

Reply:
Yes, understood.

In fact, my comment was just to mention this change but I don't know if this was better before or now with your change.